internal/controlplane/handlers_authz.go

```diff
@@ -434,28 +411,18 @@ func (s *Server) assignRole(
 	subject string,
 ) (*minder.AssignRoleResponse, error) {
 	var err error
-	// We may be given a human-readable identifier which can vary over time. Resolve
-	// it to an IDP-specific stable identifier so that we can support subject renames.
-	identity := &auth.Identity{
-		Provider:  nil,
-		UserID:    subject,
-		HumanName: subject,
-	}
-	if flags.Bool(ctx, s.featureFlags, flags.IDPResolver) {
-		identity, err = s.idClient.Resolve(ctx, subject)
-		if err != nil {
-			zerolog.Ctx(ctx).Error().Err(err).Msg("error resolving identity")
-			return nil, util.UserVisibleError(codes.NotFound, "could not find identity %q", subject)
-		}
+	// Resolve the subject to an identity
+	identity, err := s.idClient.Resolve(ctx, subject)
```

Comment:
Not sure as it might be considered similar but how about we move resolving the identity to happen after `GetUserBySubject`? This way we should try to resolve only if the caller knows the `identity_subject` of the invitee?